src/server/game/Entities/Creature/GossipDef.cpp

```diff
@@ -224,10 +224,13 @@ void PlayerMenu::SendGossipMenu(uint32 titleTextId, ObjectGuid objectGUID)
         {
             ++count;
             data << uint32(questID);
-            data << uint32(item.QuestIcon);
+            data << (quest->IsAutoComplete() ? 0 : item.QuestIcon);
```

Comment:
don't break packet structure - explicitly state the type

Reply:
also, cross-referencing master branch https://github.com/TrinityCore/TrinityCore/blob/master/src/server/game/Entities/Creature/GossipDef.cpp#L243

Gives you an idea what client expects in this packet - id, icon, level, flags, repeatable

Reply:
I see there is text.Repeatable = quest->IsRepeatable(), while this PR has a check about IsAutoComplete() too